secret/vault/update.go

```diff
@@ -66,6 +67,10 @@ func (c *Client) Update(ctx context.Context, sType, org, name string, s *api.Sec
 		vault.Data["allow_substitution"] = s.GetAllowSubstitution()
 	}
 
+	if !reflect.DeepEqual(s.GetRepoAllowlist(), sec.GetRepoAllowlist()) {
+		vault.Data["repo_allowlist"] = s.GetRepoAllowlist()
+	}
```

Comment:
i can test this if you arent sure off the top of your head.

but is this safe to make updates when no allowlist is provided, or will it revert lists back to empty because of the safe Getter?